eth/watchers/orchestratorwatcher.go

```diff
@@ -0,0 +1,141 @@
+	quit    chan struct{}
+}
+
+func NewOrchestratorWatcher(bondingManagerAddr ethcommon.Address, watcher BlockWatcher, store orchestratorStore, lpEth eth.LivepeerEthClient) (*OrchestratorWatcher, error) {
+	dec, err := NewEventDecoder(bondingManagerAddr, contracts.BondingManagerABI)
+	if err != nil {
+		return nil, err
+	}
+
+	return &OrchestratorWatcher{
+		store:   store,
+		dec:     dec,
+		watcher: watcher,
+		lpEth:   lpEth,
+		quit:    make(chan struct{}),
+	}, nil
+}
+
+// Watch starts the event watching loop
+func (ow *OrchestratorWatcher) Watch() {
+	events := make(chan []*blockwatch.Event, 10)
+	sub := ow.watcher.Subscribe(events)
+	defer sub.Unsubscribe()
+
+	for {
+		select {
+		case <-ow.quit:
+			return
+		case err := <-sub.Err():
+			glog.Error(err)
+		case events := <-events:
+			ow.handleBlockEvents(events)
+		}
+	}
+}
+
+// Stop watching for events
+func (ow *OrchestratorWatcher) Stop() {
+	close(ow.quit)
+}
+
+func (ow *OrchestratorWatcher) handleBlockEvents(events []*blockwatch.Event) {
+	for _, event := range events {
+		for _, log := range event.BlockHeader.Logs {
+			if event.Type == blockwatch.Removed {
+				log.Removed = true
+			}
+			if err := ow.handleLog(log); err != nil {
+				glog.Error(err)
+			}
+		}
+	}
+}
+
+func (ow *OrchestratorWatcher) handleLog(log types.Log) error {
+	eventName, err := ow.dec.FindEventName(log)
+	if err != nil {
+		// Noop if we cannot find the event name
+		return nil
+	}
+
+	switch eventName {
+	case "TranscoderActivated":
+		if err := ow.handleTranscoderActivated(log); err != nil {
+			return err
+		}
+	case "TranscoderDeactivated":
+		if err := ow.handleTranscoderDeactivated(log); err != nil {
+			return err
+		}
+	default:
+		return nil
+	}
+	return nil
+}
+
+func (ow *OrchestratorWatcher) handleTranscoderActivated(log types.Log) error {
+	var transcoderActivated contracts.BondingManagerTranscoderActivated
+	if err := ow.dec.Decode("TranscoderActivated", log, &transcoderActivated); err != nil {
+		return err
+	}
+
+	if !log.Removed {
+		if err := ow.store.UpdateOrch(common.NewDBOrch("", transcoderActivated.Transcoder.String(), 0, transcoderActivated.ActivationRound.Int64(), maxFutureRound)); err != nil {
+			return err
+		}
+	} else {
+		t, err := ow.lpEth.GetTranscoder(transcoderActivated.Transcoder)
+		if err != nil {
+			return err
+		}
+		if err := ow.store.UpdateOrch(common.NewDBOrch("", t.Address.String(), 0, t.ActivationRound.Int64(), t.DeactivationRound.Int64())); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (ow *OrchestratorWatcher) handleTranscoderDeactivated(log types.Log) error {
+	var transcoderDeactivated contracts.BondingManagerTranscoderDeactivated
+	if err := ow.dec.Decode("TranscoderDeactivated", log, &transcoderDeactivated); err != nil {
+		return err
+	}
+
+	if !log.Removed {
+		if err := ow.store.UpdateOrch(common.NewDBOrch("", transcoderDeactivated.Transcoder.String(), 0, 0, transcoderDeactivated.DeactivationRound.Int64())); err != nil {
```

Comment:
Could remove the `err != nil` check here and just return

Reply:
fixed in d3fe26f